### PR TITLE
Add missing package-info and relevant checkstyle rule

### PR DESCRIPTION
**Pull request by Draycia**

### CODE REVIEW

File: api/src/main/java/net/kyori/adventure/text/renderer/package-info.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Server-side component rendering.
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/text/renderer/package-info.java:25]: This doesn't *have* to be server side. I think just component rendering would suffice, perhaps with a description as to what that actually means?

File: api/src/main/java/net/kyori/adventure/translation/package-info.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Server based translations and translation registries.
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/translation/package-info.java:25]: Same here, doesn't *have* to be server-side.

File: serializer-configurate3/src/main/java/net/kyori/adventure/serializer/configurate3/package-info.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Serializers for configurate 3.
```
Comment (kezz) [inline serializer-configurate3/src/main/java/net/kyori/adventure/serializer/configurate3/package-info.java:25]: Capital C for Configurate?

**Review by kezz — APPROVED** [review]
I love pretty looking javadocs :)

**Comment by Draycia** [comment]
Looks like I missed some, I checked locally and it said all were documented properly.
Will add the missing ones shortly.

File: api/src/main/java/net/kyori/adventure/nbt/api/package-info.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Holders of Named Binary Tags.<br>
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/nbt/api/package-info.java:25]: Is the `<br>` necessary?

File: api/src/main/java/net/kyori/adventure/text/renderer/package-info.java (hunk 1 above)
Comment (Draycia) [reply api/src/main/java/net/kyori/adventure/text/renderer/package-info.java:25]: The TranslatableComponentRenderer specifies server-side which is what I went with. Will make the change, however.

File: api/src/main/java/net/kyori/adventure/nbt/api/package-info.java (hunk 4 above)
Comment (Draycia) [reply api/src/main/java/net/kyori/adventure/nbt/api/package-info.java:25]: No, will remove. It was necessary for all of 5 seconds when I initially wrote it.

File: .checkstyle/suppressions.xml [hunk 5]
```diff
@@ -7,6 +7,7 @@
   <!-- no javadoc on test classes -->
   <suppress files="src[\\/]test[\\/]java[\\/].*" checks="FilteringWriteTag"/>
   <suppress files="src[\\/]test[\\/]java[\\/].*" checks="MissingJavadoc.*"/>
+  <suppress files="src[\\/]test[\\/]java[\\/].*" checks="JavadocPackage"/>
```
Comment (kashike) [inline .checkstyle/suppressions.xml:10]: a b c d e f g h i ***j*** k l ***m*** n o...

**Review by kashike — CHANGES_REQUESTED** [review]

File: .checkstyle/suppressions.xml (hunk 5 above)
Comment (Draycia) [reply .checkstyle/suppressions.xml:10]: what's an alphabet

Comment (kashike) [reply .checkstyle/suppressions.xml:9]: you tell me
oh

**Review by zml2008 — APPROVED** [review]

File: text-serializer-plain/src/main/java/net/kyori/adventure/text/serializer/plain/package-info.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Plain string based serialization and deserialization.
```
Comment (kashike) [inline text-serializer-plain/src/main/java/net/kyori/adventure/text/serializer/plain/package-info.java:25]:
```suggestion
 * Plain text based serialization and deserialization.
```
?

**Review by MiniDigger — APPROVED** [review]

File: api/src/main/java/net/kyori/adventure/identity/package-info.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Identity information used to track the sender of messages for the social interaction
+ * features introduced in <em>Minecraft: Java Edition</em> 1.16.4.
+ */
+package net.kyori.adventure.identity;
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/identity/package-info.java:28]: Perhaps a `@sinceMinecraft` tag on this package? Not sure if makes too much sense but could get some use out of that annotation :')

File: api/src/main/java/net/kyori/adventure/nbt/api/package-info.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Holders of Named Binary Tags.
+ *
+ * @see <a href="https://wiki.vg/NBT">wiki.vg/NBT</a>
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/nbt/api/package-info.java:27]: I'd consider also linking the Minecraft Wiki page on NBT as well (or instead) if we're linking to other pages here.

File: api/src/main/java/net/kyori/adventure/permission/package-info.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * TriState based permission checking.
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/permission/package-info.java:25]:
```suggestion
 * {@link net.kyori.util.TriState} based permission checking.
```

File: api/src/main/java/net/kyori/adventure/pointer/package-info.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Key/Value based pointers to resources.
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/pointer/package-info.java:25]:
```suggestion
* Key/Value based pointers to information.
```

**Review by kezz — COMMENTED** [review]
Just a few more nitpicks :p

File: api/src/main/java/net/kyori/adventure/nbt/api/package-info.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Holders of Named Binary Tags.
+ *
+ * @see <a href="https://wiki.vg/NBT">wiki.vg/NBT</a> <a href="https://minecraft.fandom.com/wiki/NBT_format">minecraft.fandom</a>
```
Comment (zml2008) [inline api/src/main/java/net/kyori/adventure/nbt/api/package-info.java:27]:
```suggestion
 * @see <a href="https://wiki.vg/NBT">wiki.vg/NBT</a>
 * @see <a href="https://minecraft.fandom.com/wiki/NBT_format">minecraft.fandom.com/wiki/NBT_format</a>
```

incomplete thought here?

File: text-serializer-gson-legacy-impl/src/main/java/net/kyori/adventure/text/serializer/gson/legacyimpl/package-info.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * GSON legacy hover event serialization and deserialization.
```
Comment (zml2008) [inline text-serializer-gson-legacy-impl/src/main/java/net/kyori/adventure/text/serializer/gson/legacyimpl/package-info.java:25]:
```suggestion
 * Gson legacy hover event serialization and deserialization.
```

File: text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/package-info.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * GSON based serialization and deserialization.
```
Comment (zml2008) [inline text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/package-info.java:25]:
```suggestion
 * Gson based serialization and deserialization.
```

**Review by zml2008 — CHANGES_REQUESTED** [review]
few more nitpicks :p

**Review by zml2008 — APPROVED** [review]

**Pull request merged**